Выполнено ДЗ 1

mstowsky opened the pull request.

mstowsky committed:
Vesrion 0.5

Kotik:
1. Реализован параметризированный конструктор.
2. Релизовано 4 метода поведения: sleep(), play(), chaseMouse(), scratchSofa().
3. Реализован 1 метод eat() без параметров.
4. Реализован метод liveAnotherDay().

Application:
1. Создается один экземпляр Kotik (через параметризированный конструктор).
2. Вызывается метод liveAnotherDay().

mstowsky committed:
Vesrion 1.0

Общее:
1. Класс Kotik перемещен в пакет model.

Kotik:
1. Реализован счетчик количества котов.
2. Сообщение о новом созданном коте вынесено в отдельный метод.
3. В метод eat() добавлен вывод количества единиц "накормленности".
4. Релизованы геттеры и сеттеры для всех переменных класса.
5. Реализовано 3 перегруженных метода для кормления eat().
6. Реализован вывод текущего часа в формате "hh:mm".

Application:
1. Реализован вывод количества созданных котов.
2. Реализован вывод сообщения о новом созданном коте.
3. Создаются два кота, у каждого выводится распорядок дня.
4. Реализован вывод сравнения их разговора.

Kuznetsov-DI commented on .gitignore:
```diff
@@ -1 +1,2 @@
 /.idea/
```
Эта папка должна быть в gitignore

mstowsky replied:
Вроде поправил, будет понятно после принятия pull request'а.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+
+public class Kotik {
+    private static int kotikCount = 0;
+    private String name;
```
обычно для наглядности делают отступ между статическими и обычными переменными

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    private String meow;
+
+    private static void encreaseCounter() {
+        kotikCount += 1;
```
kotikCount++;

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    private int isNotHungry; // 0 - чем ближе к нулю, тем кот голоднее
+    private String meow;
+
+    private static void encreaseCounter() {
```
Методы обычно в порядке модификаторов пишутся, от public до private, public сверху

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    // сеттеры и геттеры для параметров класса model.Kotik:
+    public void setName(String name) {this.name = name;}
```
между методами всегда отступ, и не пиши в одну строку, так не делают, прожимай ctrl+alt+L всегда, после того как с классом поработал

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    private static int kotikCount = 0;
+    private String name;
+    private double weight;
+    private int isNotHungry; // 0 - чем ближе к нулю, тем кот голоднее
```
префикс is пишется перед булевой переменной у интовой такого быть не должно. Самый простой способ нейминга вот)
![image](https://github.com/mstowsky/QAA-training-popov-artem/assets/48683598/cfadd6ee-faae-43c3-9330-5171f25ad7b6)

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    // параметризированный конструктор объекта:
+    public Kotik(String name, double weight, String meow) {
```
Конструкторы всегда идут над методами.
Т.е. порядок:
1. Переменные (в порядке модификаторов доступа)
2. Конструкторы
3. Методы (в порядке модификаторов доступа)

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    // сеттеры и геттеры для параметров класса model.Kotik:
+    public void setName(String name) {this.name = name;}
+    public void setWeight(double weight) {this.weight = weight;}
+    public void setIsHungry() {this.isNotHungry = (int) (Math.random() * 10);} // степень голода устанавливается случайным числом от 0 до 10;
```
Это не правильная реализация сеттера, его суть в том что он принимает параметр и устанавливает его.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    public int getIsNotHungry() {return isNotHungry;}
+    public String getMeow() {return meow;}
+
+    public void newCatHere() {
```
метод всегда в названии обозначает действие, т.е. глагол.
printNewCatInfo() типо того для твоего метода

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+        System.out.println("We have new cat here! His name is " + name + ", he weights " + weight + " kg, he is satisfied (not hungry) for " + isNotHungry + ".");
+    }
+
+    private void wantToEat(String cantDoAction) { // вывод сообщения о голодном коте
```
Тоже название немного не клеится, в комменте написано, что это вывод сообщения...

Название параметра тоже сомнительное, почему не просто action?

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    // 5 дел для распорядка дня:
+    public void play() {
+        System.out.println(name + " is playing with toys.");
+        isNotHungry -= 1;
```
просто --.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    // 3 перегруженных метода для кормления:
+    public void eat(int isNotHungry, String meal) {
```
![image](https://github.com/mstowsky/QAA-training-popov-artem/assets/48683598/4521c32b-75c1-4cd1-a0b6-22b6e88cb8ac)

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+        System.out.println("He took his meal and now is satisfied for " + isNotHungry + ".");
+    }
+    public void eat() {
+        eat(1 + (int) (Math.random() * 5), "Sheba"); // покормить кота на 1-5 единиц
```
Обычно перегрузку делают следующим образом.

У тебя полную реализацию имеет один метод с наибольшим количеством параметров.
Остальные, вызывают его внутри себя, передавая захардкоженные параметры или null. Что бы в случае изменения править в одном месте а не везде, если логика методов особо не отличается.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,119 @@
+            System.out.print(currentHour + ":00 ");
+            switch (caseRandom) {
+                case 1:
+                    if (isNotHungry > 0) play();
```
У тебя 5 раз проверяется 1 и тот же if, может лучше его вынести на уровень вышел, до switch, тогда достаточно будет 1.

mstowsky replied:
Логика такая: выбирается действие -> проверяется уровень сытости -> если голодный, то действие не делаю, а прошу поесть. В текущей реализации выводится информация какое именно действие отказался выполнять kotik. Решил оставить так.

mstowsky committed:
Vesrion 1.1

Исправил замечания первого ревью.
Удалил папку .idea и выполнил rebuild project.
Сделал вызов liveAnotherDay() только для одного созданного кота вместо двух.

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    // 3 перегруженных метода для кормления:
+    public void eat(int satietyLevel, String meal) {
```
Все правильно, только наоборот расставь, который юзается ниже, а который юзает сверху. Так удобнее читать

Kuznetsov-DI commented on homework-1/src/model/Kotik.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    // вывод сообщения о голодном коте:
+    private void printHungryCatMessage(String action) {
```
print... and Feed, там вроде еще и кормишь ты.

mstowsky replied:
Нет, в этом методе только вывод сообщения. Кормим методом eat().

mstowsky committed:
Поправил порядок перегружаемых методов
